src/tools/generators.ts

```diff
@@ -335,30 +389,42 @@ export async function validateAppIconGenerator(option: `${Platforms}` | "all") {
 
   // validate both presence and size of all input files
   const validationPromises = inputFileNames.map(async (fileName) => {
-    const filePath = path(templatesDir(), "app-icon", fileName)
+    const boilerplateInputFilePath = path(sourceDirectory(), "app-icon", fileName)
+    const inputFilePath = path(templatesDir(), "app-icon", fileName)
 
-    const isMissing = !exists(filePath)
+    const isMissing = !exists(inputFilePath)
     const isInvalidSize = await (async function () {
       if (isMissing) return false
 
-      const metadata = await sharp(filePath).metadata()
+      const metadata = await sharp(inputFilePath).metadata()
       return metadata.width !== 1024 || metadata.height !== 1024
     })()
+    const isSameAsSource = await (async function () {
+      if (skipSourceEqualityValidation) return false
+      if (isMissing) return false
+
+      const inputFileMd5 = inspect(inputFilePath, { checksum: "md5" }).md5
+      const sourceFileMd5 = inspect(boilerplateInputFilePath, { checksum: "md5" }).md5
+
+      return inputFileMd5 === sourceFileMd5
+    })()
```

Comment:
adds a file signature check here to notify the user if a template input-file has been unchanged. Allows to force skip this check if a flag is passed in.